src/Decompiler/Typing/ExpressionTypeAscender.cs

```diff
@@ -274,8 +278,10 @@ public DataType VisitFieldAccess(FieldAccess acc)
             throw new NotImplementedException();
         }
 
-        public DataType VisitIdentifier(Identifier id)
+        public virtual DataType VisitIdentifier(Identifier id)
         {
```

Comment:
New implementation does not require making `VisitIdentifier` virtual. See 93dd31e